MarkupsToModel/MRML/vtkMRMLMarkupsToModelNode.cxx

```diff
@@ -214,48 +217,26 @@ void vtkMRMLMarkupsToModelNode::PrintSelf( ostream& os, vtkIndent indent )
 {
   vtkMRMLNode::PrintSelf(os,indent); // This will take care of referenced nodes
   os << indent << "ModelID: ";
-  //for (int i=0; i<this->GetNumberOfTools();i++)
-  //{
-    //os << indent << this->Markups->GetID() << std::endl;
-  //}
-  
 }
 
 vtkMRMLMarkupsFiducialNode * vtkMRMLMarkupsToModelNode::GetMarkupsNode()
 {
-  //std::list<MarkupsTool>::iterator it = this->Markups.begin();
-  //advance (it,currentRow);
-  //MarkupsTool * watchedTool = &(*it);
-  vtkMRMLMarkupsFiducialNode* markupsNode = vtkMRMLMarkupsFiducialNode::SafeDownCast( this->GetNodeReference( MARKUPS_ROLE ) );
+  vtkMRMLMarkupsFiducialNode* markupsNode = vtkMRMLMarkupsFiducialNode::SafeDownCast( this->GetNodeReference( INPUT_MARKUPS_ROLE ) );
   return markupsNode;
-  //return this->Markups;
-}
-
-//void vtkMRMLMarkupsToModelNode::AddMarkupsNode( vtkMRMLMarkupsFiducialNode* markupsAdded)
-//{
-//  //MarkupsTool tempMarkupsTool;
-//  if(markupsAdded==NULL)
-//  {
-//    return;
-//  }
-//
-//  this->Markups=markupsAdded;
-//  //tempMarkupsTool.tool=markupsAdded;
-//  //tempMarkupsTool.label=getToolLabel(markupsAdded->GetName());
-//  //tempMarkupsTool.id=markupsAdded->GetID();
-//  ////tempMarkupsTool.LastTimeStamp=mrmlNode->GetMTime();
-//  //this->Markups.push_back(tempMarkupsTool);
-//
-//  //return GetNumberOfTools();
-//}
+}
+
+vtkMRMLModelNode * vtkMRMLMarkupsToModelNode::GetModelNode()
+{
+  vtkMRMLModelNode* modelNode = vtkMRMLModelNode::SafeDownCast( this->GetNodeReference( OUTPUT_MODEL_ROLE ) );
+  return modelNode;
+}
 
 void vtkMRMLMarkupsToModelNode::SetAndObserveMarkupsNodeID( const char* markupsId )
 {
   // SetAndObserveNodeReferenceID does not handle nicely setting of the same
   // node (it should simply ignore the request, but it adds another observer instead)
   // so check for node equality here.
```

Comment:
I meant that we should check if this is still true:
// SetAndObserveNodeReferenceID does not handle nicely setting of the same  
// node (it should simply ignore the request, but it adds another observer instead)  
// so check for node equality here. 

But I checked the code and actually it is useful to check if the node is changed or not also because it's better not to invoke InputDataModifiedEvent if there is no change. So, just remove this 3-line comment.